WooCommerce/WooCommerceTests/Notifications/PushNotificationsManagerTests.swift

```diff
@@ -346,10 +475,55 @@ final class PushNotificationsManagerTests: XCTestCase {
         let emittedNotification = emittedNotifications.first!
         XCTAssertEqual(emittedNotification.kind, .storeOrder)
         XCTAssertEqual(emittedNotification.noteID, 9_981)
+        XCTAssertEqual(emittedNotification.title, Sample.defaultMessage)
+        XCTAssertNil(emittedNotification.subtitle)
+        XCTAssertNil(emittedNotification.message)
+    }
+
+    func test_it_emits_foreground_notifications_when_it_receives_a_notification_while_app_is_active() throws {
+        // Given
+        application.applicationState = .active
+
+        let featureFlagService = MockFeatureFlagService(isPushNotificationsForAllStoresOn: true)
+        manager = {
+            let configuration = PushNotificationsConfiguration(application: self.application,
+                                                               defaults: self.defaults,
+                                                               storesManager: self.storesManager,
+                                                               supportManager: self.supportManager,
+                                                               userNotificationsCenter: self.userNotificationCenter)
+
+            return PushNotificationsManager(configuration: configuration, featureFlagService: featureFlagService)
+        }()
+
+        var emittedNotifications = [PushNotification]()
+        _ = manager.foregroundNotifications.subscribe { notification in
+            emittedNotifications.append(notification)
+        }
+
+        let userinfo = notificationPayload(noteID: 9_981,
+                                           type: .storeOrder,
+                                           title: Sample.defaultTitle,
+                                           subtitle: Sample.defaultSubtitle,
+                                           message: Sample.defaultMessage,
+                                           featureFlagService: featureFlagService)
+
+        // When
+        manager.handleNotification(userinfo, onBadgeUpdateCompletion: {}) { _ in
+            // noop
+        }
+
+        // Then
+        XCTAssertEqual(emittedNotifications.count, 1)
+
+        let emittedNotification = emittedNotifications.first!
```

Comment:
Maybe `XCTUnwrap` would be better. 🙂 

```suggestion
        let emittedNotification = try XCTUnwrap(emittedNotifications.first)
```

Reply:
Nice catch! Updated in https://github.com/woocommerce/woocommerce-ios/pull/5299/commits/eea115f296e1ab8e5f8203d12e2c479518542469